src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantAdvertService.java

```diff
@@ -84,4 +90,21 @@ public GrantAdvert getAdvertBySchemeId(String schemeId) {
         log.debug("Advert with schemeId {} found", schemeId);
         return grantAdvert;
     }
+
+    public GetGrantAdvertDto grantAdvertToDto(
+            final GrantAdvert grantAdvert,
+            final String sub,
+            final Integer schemeId
+    ) {
+        final GrantApplicant grantApplicant = grantApplicantService.getApplicantById(sub);
+        GetGrantMandatoryQuestionDto mandatoryQuestionsDto = null;
+
+        if (grantMandatoryQuestionService.existsBySchemeIdAndApplicantId(schemeId, grantApplicant.getId())) {
+            final GrantMandatoryQuestions grantMandatoryQuestions = grantMandatoryQuestionService
+                    .getMandatoryQuestionBySchemeId(schemeId, sub);
+            mandatoryQuestionsDto = grantMandatoryQuestionMapper.mapGrantMandatoryQuestionToGetGrantMandatoryQuestionDTO(grantMandatoryQuestions);
+        }
+
+        return generateGetGrantAdvertDto(grantAdvert, mandatoryQuestionsDto);
```

Comment:
What happens if mandatoryQuestionsDto is null, is this an expected scenario?

Reply:
Yeah in all cases this is used, doesn't matter if thats null or not